config/vite/plugin-mastodon-themes.ts

```diff
@@ -0,0 +1,58 @@
+/* This plugins handles Mastodon's theme system
+ */
+
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+import yaml from 'js-yaml';
+import type { Plugin, UserConfig } from 'vite';
+
+export function MastodonThemes(): Plugin {
+  return {
+    name: 'mastodon-themes',
+    config: async (userConfig: UserConfig) => {
```

Comment:
this format doesn't require specifying the type, as it comes from `Plugin`.
```suggestion
    async config(userConfig) {
```